mmv1/templates/terraform/custom_create/firebase_app_check_service_config.go

```diff
@@ -0,0 +1,62 @@
+userAgent, err := tpgresource.GenerateUserAgentString(d, config.UserAgent)
+if err != nil {
+	return err
+}
+
+obj := make(map[string]interface{})
+enforcementModeProp, err := expandFirebaseAppCheckServiceConfigEnforcementMode(d.Get("enforcement_mode"), d, config)
+if err != nil {
+	return err
+} else if v, ok := d.GetOkExists("enforcement_mode"); !tpgresource.IsEmptyValue(reflect.ValueOf(enforcementModeProp)) && (ok || !reflect.DeepEqual(v, enforcementModeProp)) {
+	obj["enforcementMode"] = enforcementModeProp
+}
+log.Printf("[DEBUG] Creating new ServiceConfig: %#v", obj)
+
+project, err := tpgresource.GetProject(d, config)
+if err != nil {
+	return fmt.Errorf("Error fetching project for ServiceConfig: %s", err)
+}
+billingProject := project
+
+url, err := tpgresource.ReplaceVars(d, config, "{{FirebaseAppCheckBasePath}}projects/{{project}}/services/{{service_id}}")
+if err != nil {
+	return err
+}
+
+// Custom logic: add all mutable fields to the updateMask
+url, err = transport_tpg.AddQueryParams(url, map[string]string{"updateMask": "enforcementMode"})
```

Comment:
We should be able to append the update mask to the create and delete urls in order to avoid needing to use `custom_x` methods here. They're pretty heavyweight and can be hard to spot the difference between generated and custom/handwritten code, so we avoid them as much as possible! https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/products/appengine/ApplicationUrlDispatchRules.yaml is an example where we've done this previously.

Reply:
I forgot this old trick. Yes, this saves lots of code.